### DISCUSSION TITLE
How define route with parameter?

### DISCUSSION THREAD

**Discussion by krajcikondra** [GeniusesOfSymfony/WebSocketBundle#460] [Q&A]
Hi,

I chat RoomChatTopic with routing.yaml file

```
roomChat_topic:
  pattern: roomChat/channel
  callback: 'roomChat.topic'
```

But I want to have more then one one room. Therefor I need distinguish room where user connect. Parameter in routing parttern seems me to be good solution. But I am not sure if is possible define parameter in pattern like this:

```
roomChat_topic:
  pattern: roomChat/channel/{$roomId}
  callback: 'roomChat.topic'
```

When I define routing pattern with parameter and try connect by javascript client

`  myWs.subscribe('roomChat/channel/123', (uri, msg) => { ... });`

I get following error:

`10:06:35 ERROR     [websocket] Unable to find route for roomChat/channel/123`

It is some other way to define parameter?

**Comment by mbabker** [comment]
https://github.com/GeniusesOfSymfony/WebSocketBundle/blob/3.x/docs/rpc.md#step-3-add-a-route-for-your-service has an example of a param in the route.  Essentially, it’s the same as configuring a normal Symfony router with YAML.
